core-bundle/src/Resources/contao/models/FilesModel.php

```diff
@@ -394,6 +395,40 @@ public static function findMultipleFoldersByFolder($strPath, array $arrOptions=a
 	protected function postSave($intType)
 	{
 	}
+
+	/**
+	 * Get the meta fields defined in `tl_files.meta.eval.metaFields`.
+	 */
+	public static function getMetaFields(): array
+	{
+		Controller::loadDataContainer('tl_files');
+
+		return array_keys($GLOBALS['TL_DCA']['tl_files']['fields']['meta']['eval']['metaFields'] ?? array());
+	}
+
+	/**
+	 * Get the meta data for this file. Specify one or more locales - meta
+	 * data of the first matching one will be returned or null if none was
+	 * found.
+	 */
+	public function getMetaData(string ...$locales): ?MetaData
+	{
+		$dataCollection = StringUtil::deserialize($this->meta, true);
+
+		foreach ($locales as $locale)
+		{
+			$data = $dataCollection[$locale] ?? array();
+
+			if (!empty($data))
+			{
+				// todo: should we strip additional fields here as well or is the serialized
+				//       array considered to be 'in line' with the current config?
```

Comment:
I think we should not strip something here. If someone extends the meta data wizard in a dynamic manner, we should not remove the data here IMO.